cms/djangoapps/contentstore/views/transcript_settings.py

```diff
@@ -108,3 +119,138 @@ def transcript_credentials_handler(request, course_key_string):
             response = JsonResponse({'error': error_message}, status=400)
 
     return response
+
+
+@login_required
+@require_GET
+def transcript_download_handler(request, course_key_string):
+    """
+    JSON view handler to download a transcript.
+
+    Arguments:
+        request: WSGI request object
+        course_key_string: course key
+
+    Returns:
+        - A 200 response with SRT transcript file attached.
+        - A 400 if there is a validation error.
+        - A 404 if there is no such transcript or feature flag is disabled.
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not VideoTranscriptEnabledFlag.feature_enabled(course_key):
+        return HttpResponseNotFound()
+
+    missing = [attr for attr in ['edx_video_id', 'language_code'] if attr not in request.GET]
+    if missing:
+        return JsonResponse(
+            {'error': _(u'Following parameters are required: {missing}.').format(missing=', '.join(missing))},
+            status=400
+        )
+
+    edx_video_id = request.GET['edx_video_id']
+    language_code = request.GET['language_code']
+    transcript = get_video_transcript_data(video_ids=[edx_video_id], language_code=language_code)
+    if transcript:
+        name_and_extension = os.path.splitext(transcript['file_name'])
+        basename, file_format = name_and_extension[0], name_and_extension[1][1:]
+        transcript_filename = '{base_name}.{ext}'.format(base_name=basename.encode('utf8'), ext=Transcript.SRT)
+        transcript_content = Transcript.convert(
+            content=transcript['content'],
+            input_format=file_format,
+            output_format=Transcript.SRT
+        )
+        # Construct an HTTP response
+        response = HttpResponse(transcript_content, content_type=Transcript.mime_types[Transcript.SRT])
+        response['Content-Disposition'] = 'attachment; filename="{filename}"'.format(filename=transcript_filename)
+    else:
+        response = HttpResponseNotFound()
+
+    return response
+
+
+def validate_transcript_upload_data(data, files):
+    """
+    Validates video transcript file.
+    Arguments:
+        data: A request's data part.
+        files: A request's files part.
+    Returns:
+        None or String
+        If there is error returns error message otherwise None.
+    """
+    error = None
+    # Validate the must have attributes - this error is unlikely to be faced by common users.
+    must_have_attrs = ['edx_video_id', 'language_code', 'new_language_code']
+    missing = [attr for attr in must_have_attrs if attr not in data]
+    if missing:
+        error = _(u'Following parameters are required: {missing}.').format(missing=', '.join(missing))
```

Comment:
Sylvia can you look at this string - This is mostly like not user-facing string?